.github/workflows/core-application-e2e-tests-nightly.yml

```
@@ -0,0 +1,147 @@
+        run: |
+          DATABASE=elasticsearch docker compose up -d tasklist operate
+        working-directory: qa/core-application-e2e-test-suite/config
+
+      - name: List running Docker containers
+        run: docker ps -a
+        working-directory: qa/core-application-e2e-test-suite/config
+
+      - name: Wait for services to be ready
+        id: wait-for-services
+        run: |
+          echo "Checking if services are up..."
+          ready=false
+          for i in {1..90}; do
+            if ! nc -zv localhost 8080; then
+              echo "Port 8080 is not open yet..."
+            fi
+
+            if ! nc -zv localhost 8081; then
+              echo "Port 8081 is not open yet..."
+            fi
+
+            # Check if tasklist page is available (adjust the URL based on your service)
+            response_8080=$(curl -s -m 5 http://localhost:8080/tasklist || echo "Failed")
+            response_8081=$(curl -s -m 5 http://localhost:8081/operate || echo "Failed")
+
+            # Check if tasklist is up (you can adjust this condition based on actual response)
+            if [[ "$response_8080" != "Failed" ]] && [[ "$response_8081" != "Failed" ]]; then
+              echo "Services are ready!"
+              ready=true
+              break
+            fi
+
+            echo "Waiting for services... ($i/90)"
+            echo "Response from 8080: $response_8080"
+            echo "Response from 8081: $response_8081"
+            sleep 10
+          done
+
+          if [ "$ready" == "true" ]; then
+            echo "ready=true" >> $GITHUB_OUTPUT
+          else
+            echo "Services failed to start in time."
+            exit 1
+          fi
+
+      - name: Print Docker logs before failing
+        if: failure()
+        run: |
+          docker-compose logs tasklist
+          docker-compose logs operate
+        working-directory: qa/core-application-e2e-test-suite/config
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: npm
+          cache-dependency-path: qa/core-application-e2e-test-suite/package-lock.json
+
+      - name: Install dependencies
+        run: npm ci
+        working-directory: qa/core-application-e2e-test-suite
+
+      - name: Import Secrets
+        id: secrets
+        uses: hashicorp/vault-action@7709c609789c5e27b757a85817483caadbb5939a
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          exportEnv: false # we rely on step outputs, no need for environment variables
+          secrets: |
+            secret/data/github.com/organizations/camunda TESTRAIL_QA_EMAIL;
+            secret/data/github.com/organizations/camunda TESTRAIL_QA_PSW;
+
+      - name: Python setup
+        if: always()
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
```

I think we can try to cache the Python deps

```suggestion
          python-version: "3.x"
          cache: "pip"
```